examples/nanoChatGPT/train_rlhf.py

```diff
@@ -64,32 +66,38 @@ def forward(self, tensordict):
 
     # ######## TRAINING LOOP ########
 
-    # def get_action(td):
-    #     critic(td)
-    #     actor(td)
-    #     td["sample_log_prob"] = td["sample_log_prob"].detach()
-    #     return td
-
-    for i in range(config["max_iters"]):
+    ep_length = config["episode_length"]
+    max_iters = config["max_iters"]
+    num_epochs = config["num_epochs"]
+    device = config['device']
+    rb = TensorDictReplayBuffer(
+        storage=LazyTensorStorage(ep_length * config["batch_size"]),
```

Comment:
@vmoens can you explain these buffer settings?

Reply:
sure
We want a storage that is as big as the number of steps collected, such that when we get a rollout and put it in the buffer it erases the previous rollout completely.
We use a sampler without replacement such that we can iterate over the buffer and collect samples without replacement (consuming the buffer, so to say).
we set the batch size of the buffer to some value that determines how many samples we feed to the loss.